e2e/vm/support_bundle_test.go

```diff
@@ -0,0 +1,159 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package vm
+
+import (
+	"archive/zip"
+	"fmt"
+	"os"
+	"path"
+	"strings"
+	"time"
+
+	"github.com/onsi/ginkgo/v2"
+	"github.com/onsi/gomega"
+	"github.com/runfinch/common-tests/command"
+	"github.com/runfinch/common-tests/option"
+)
+
+var testSupportBundle = func(o *option.Option) {
+	ginkgo.Describe("Support bundles", func() {
+		ginkgo.It("Should generate a support bundle", func() {
+			command.Run(o, "support-bundle", "generate")
+			entries, err := os.ReadDir(".")
+			gomega.Expect(err).Should(gomega.BeNil())
+			bundleExists := false
+			for _, dirEntry := range entries {
+				if strings.Contains(dirEntry.Name(), "finch-support") {
+					_, err := os.Stat(dirEntry.Name())
+					if err == nil {
+						bundleExists = true
+					}
+
+					err = os.Remove(dirEntry.Name())
+					gomega.Expect(err).Should(gomega.BeNil())
+				}
+			}
+			gomega.Expect(bundleExists).Should(gomega.BeTrue())
+		})
+		ginkgo.It("Should generate a support bundle with an extra file included", func() {
+			includeFilename := fmt.Sprintf("tempTestfile%s", time.Now().Format("20060102150405"))
+			_, err := os.Create(includeFilename)
+			gomega.Expect(err).Should(gomega.BeNil())
+			defer func() {
+				err := os.Remove(includeFilename)
+				gomega.Expect(err).Should(gomega.BeNil())
+			}()
+
+			command.Run(o, "support-bundle", "generate", "--include", includeFilename)
+			entries, err := os.ReadDir(".")
+			gomega.Expect(err).Should(gomega.BeNil())
+			bundleExists := false
+			for _, dirEntry := range entries {
+				if strings.Contains(dirEntry.Name(), "finch-support") {
+					_, err := os.Stat(dirEntry.Name())
+					if err == nil {
+						bundleExists = true
+					}
+
+					reader, err := zip.OpenReader(dirEntry.Name())
+					gomega.Expect(err).Should(gomega.BeNil())
+
+					zipBaseName := path.Base(dirEntry.Name())
+					zipPrefix := strings.TrimSuffix(zipBaseName, path.Ext(zipBaseName))
+					_, err = reader.Open(path.Join(zipPrefix, "misc", includeFilename))
+					gomega.Expect(err).Should(gomega.BeNil())
+
+					err = os.Remove(dirEntry.Name())
+					gomega.Expect(err).Should(gomega.BeNil())
+				}
+			}
+			gomega.Expect(bundleExists).Should(gomega.BeTrue())
+		})
+		ginkgo.It("Should generate a support bundle with a default file excluded", func() {
+			command.Run(o, "support-bundle", "generate", "--exclude", "serial.log")
```

Comment:
Also add abstract and relative path for both include and exclude? It should be a quick change that only needs copy-paste

Reply:
@sam-berning How about exclude by relative path?

Reply:
Also how about invalid path for include and exclude?